Sources/Hummingbird/Middleware/MiddlewareModule/_SpreadMiddleware.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A middleware that can handle an array of middleware.
+///
+/// This middleware is useful for situations where you want to compose an array of middleware together.
+///
+/// You won't typically construct this middleware directly, but instead will use standard `for` loop
+/// statements in a result builder to automatically build spread middleware:
+///
+/// ```swift
+/// router.addMiddleware {
+///   for logger in loggers {
+///     LoggingMiddleware(logger: logger)
+///   }
+/// }
+/// ```
+public struct _SpreadMiddleware<M0: MiddlewareProtocol>: MiddlewareProtocol {
```

Comment:
Can you add `@_documentation(visibility: internal)`. This isn't a symbol we need to publish documentation for.